zipline/assets/assets.py

```
@@ -1297,133 +1279,110 @@ def _lookup_generic_scalar(self,
         On success, append to matches.
         On failure, append to missing.
         """
-        if isinstance(asset_convertible, Asset):
-            matches.append(asset_convertible)
+        if isinstance(obj, (Asset, ContinuousFuture)):
+            matches.append(obj)
+            return
 
-        elif isinstance(asset_convertible, Integral):
+        if isinstance(obj, Integral):
             try:
-                result = self.retrieve_asset(int(asset_convertible))
+                result = self.retrieve_asset(int(obj))
+                matches.append(result)
             except SidsNotFound:
-                missing.append(asset_convertible)
-                return None
-            matches.append(result)
+                missing.append(obj)
+            finally:
+                return
```

this masks any exceptions in that block